common/ayon_common/distribution/addon_distribution.py

```diff
@@ -312,185 +361,694 @@ def _try_convert_to_server_source(addon, source):
     if not filename:
         return source
 
-    return ServerResourceSource(
+    return ServerSourceInfo(
         type=UrlType.SERVER.value, filename=filename
     )
 
 
-def update_addon_state(
-    addon_infos,
-    destination_folder,
-    factory,
-    log=None
-):
-    """Loops through all 'addon_infos', compares local version, unzips.
+class DistributeTransferProgress:
```

Comment:
I don't understand what this is, shouldn't this inherit from `ayon_api.TransferProgress`?

I don't particularly like that 'Progress', I'll always imagine something like 0-100, not if if started, failed. I would prefer something like `DistributionItemStatus`

Reply:
This object should help to track the progress of single distribution item source (progress not state). The progress consist of 3 parts: "transfer", "hashcheck", "unzip". All of these steps may take some time but may not have ability to track exact 0-100. Since it was created to have ability to use it in (future) Distribution UI it does not matter, important is knowledge in which part of progress we are and 0-100 is a bonus.

Reply:
I agree with @kalisp - even from your extra explanation I feel this makes more sense as a status than progress?

Reply:
Also, looking at the actual data that this hold? Should this maybe be brought back to being an explicit dataclass?

Reply:
We were discussing it and there may be confusion in state and status. But the name is not right. It should be something like `DistributeSourceProgress` or `DistributeSourceStatus`. I have opinion that it is progress but both of you had same idea to I'm willing to loose my opinion here.

> Should this maybe be brought back to being an explicit dataclass?

I personally don't like dataclasses as it's not clear what should be read-only.

I'm merging current state as we have to start with "prove of concept tests", but we can continue with comments here, it's not 100% finished anyway and there will be more PRs.